packages/headless/src/state/commerce-app-state.ts

```diff
@@ -0,0 +1,31 @@
+import {
+  CategoryFacetSearchSection,
+  CategoryFacetSection,
+  ConfigurationSection,
+  ContextSection,
+  DateFacetSection,
+  FacetOptionsSection,
+  FacetOrderSection,
+  FacetSearchSection,
+  FacetSection,
+  NumericFacetSection,
+  PaginationSection,
+  ProductListingV2Section,
+  StructuredSortSection,
+  VersionSection,
+} from './state-sections';
+
+export type ProductListingV2AppState = ConfigurationSection &
+  ProductListingV2Section &
+  FacetSearchSection &
+  FacetSection &
+  NumericFacetSection &
+  CategoryFacetSection &
+  CategoryFacetSearchSection &
+  DateFacetSection &
+  FacetOptionsSection &
+  FacetOrderSection &
+  StructuredSortSection &
+  PaginationSection &
+  VersionSection &
+  ContextSection;
```

Comment:
- `FacetSearchSection` and `CategoryFacetSearchSection` can be removed since we don't support facet search.
- `ContextSection` can be removed since we won't support untyped context.
- `FacetOptionsSection` can be removed since these options will only be available to merchandisers, through the platform.

```suggestion
export type ProductListingV2AppState = ConfigurationSection &
  ProductListingV2Section &
  FacetSection &
  NumericFacetSection &
  CategoryFacetSection &
  DateFacetSection &
  FacetOrderSection &
  StructuredSortSection &
  PaginationSection &
  VersionSection;
```

This impacts the `StateNeededByFetchProductListingV2` as well!